docs/DafnyRef/Expressions.md

```diff
@@ -463,17 +466,20 @@ Right-hand-side expressions appear in the following constructs:
 These are the only contexts in which arrays or objects may be
 allocated, or in which havoc may be produced.
 
-## 22.16. Array Allocation
+## 21.16. Array Allocation
 ````grammar
-ArrayAllocation_ = "new" Type "[" Expressions "]"
+ArrayAllocation_ = "new" [ Type ] "[" Expressions "]"
```

Comment:
This `Expressions` is also optional. But if it's omitted, then the `"["` alternative below must be taken.